conda/notices/core.py

```diff
@@ -35,69 +36,104 @@ def display_notices(
         silent: Whether to use a spinner when fetching and caching notices.
     """
     channel_name_urls = get_channel_name_and_urls(get_channel_objs(context))
-    channel_notice_responses = http.get_notice_responses(channel_name_urls, silent=silent)
+    channel_notice_responses = fetch.get_notice_responses(channel_name_urls, silent=silent)
     channel_notices = flatten_notice_responses(channel_notice_responses)
-    num_total_notices = len(channel_notices)
+    total_number_channel_notices = len(channel_notices)
 
     cache_file = cache.get_notices_cache_file()
+
+    # We always want to modify the mtime attribute of the file if we are trying to retrieve notices
+    # This is used later in "is_channel_notices_cache_expired"
+    cache_file.touch()
+
     viewed_notices = None
-    num_viewed_notices = 0
+    viewed_channel_notices = 0
     if not always_show_viewed:
         viewed_notices = cache.get_viewed_channel_notice_ids(cache_file, channel_notices)
-        num_viewed_notices = len(viewed_notices)
+        viewed_channel_notices = len(viewed_notices)
 
     channel_notices = filter_notices(channel_notices, limit=limit, exclude=viewed_notices)
-    if len(channel_notices) == 0:
-        return
 
-    views.print_notices(channel_notices)
+    return ChannelNoticeResultSet(
+        channel_notices=channel_notices,
+        viewed_channel_notices=viewed_channel_notices,
+        total_number_channel_notices=total_number_channel_notices,
+    )
+
+
+def display_notices(channel_notice_set: ChannelNoticeResultSet) -> None:
+    """
+    Prints the channel notices to std out
+    """
```

Comment:
If we offer a default, we can avoid the change in `main_notices.py`:
```suggestion
def display_notices(channel_notice_set: ChannelNoticeResultSet | None = None) -> None:
    """
    Prints the channel notices to std out
    """
    if channel_notice_set is None:
        channel_notice_set = retrieve_notices()
```

Reply:
I understand the desire for this change, but for me, this violates the, "separation of concerns" and "loose coupling" principles that I am trying to follow. A display function should not also perform a retrieval. It should merely define an interface for accepting a data structure that you would like displayed.

I hope you understand where I am coming from.

Reply:
+1 to @travishathaway I think that composition patterns like this are more effective than subtle defaults that make code harder to read.